src/modules/onboarding.ts

```diff
@@ -77,6 +77,39 @@ export default class Onboarding {
     });
   }
 
+  /**
+   * Sign the 'key derivation' onboarding message to receive a signature.
+   *
+   * @param ethereumAddress of the account
+   * @param signingMethod Method to use for signing
+   * @returns Signature used to derive your STARK key pairs
+   */
+  protected async signStarkKeyDerivationMessage(
+    ethereumAddress: string,
+    signingMethod: SigningMethod = SigningMethod.TypedData,
+  ): Promise<string> {
+    if (!KEY_DERIVATION_SUPPORTED_SIGNING_METHODS.includes(signingMethod)) {
+      throw new Error('Unsupported signing method for API key derivation');
+    }
+
+    const message: OnboardingAction = {
+      action: OnboardingActionString.KEY_DERIVATION,
+    };
+
+    // On mainnet, include an extra onlySignOn parameter.
+    if (this.networkId === 1) {
+      message.onlySignOn = 'https://trade.dydx.exchange';
+    }
+
+    const signature = await this.signer.sign(
+      ethereumAddress,
+      signingMethod,
+      message,
+    );
+
+    return signature;
```

Comment:
nit: either give a type to `signature` or just do `return this.signer.sign(...);`